### PR TITLE
add noarch dep message

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@conda-forge/staged-recipes @conda-forge/core This implements half of https://github.com/conda-forge/staged-recipes/issues/4787

**Comment by scopatz** [comment]
I am :+1: for this

**Comment by jakirkham** [comment]
Probably should point out selectors are still not allowed in `noarch` recipes.

**Comment by jakirkham** [comment]
Also feel free to add `[ci skip]` in the commit messages for things like this.

File: recipes/example/meta.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ source:
 
 build:
   # Uncomment the following line if the package is pure python and the recipe is exactly the same for all platforms.
+  # It is ok if the dependencies are not built for all platforms/versions, although selectors are still not allowed.
```
Comment (dougalsutherland) [inline recipes/example/meta.yaml]: Nit: maybe replace `ok` with `okay` for the pedants; default-English spelling uses either `okay` or `OK`/`O.K.` ([source](https://english.stackexchange.com/questions/25617/difference-between-ok-and-okay)).

Comment (CJ-Wright) [reply recipes/example/meta.yaml]: Relvent: https://www.youtube.com/watch?v=ytlNgZMpyeY

**Pull request merged**